pkg/kubectl/cmd/config/view.go

```diff
@@ -106,6 +109,11 @@ func (o ViewOptions) Run(out io.Writer, printer kubectl.ResourcePrinter) error {
 		}
 	}
 
+	if o.Short {
+		fmt.Printf("short\n")
```

Comment:
It's intentional.  For templates, you won't it shortened.  If you're shortening, then the output is no longer a useable kubeconfig, so you don't want it to parse.